pkg/services/datasources/service/datasource.go

```diff
@@ -482,6 +483,26 @@ func (s *Service) httpClientOptions(ctx context.Context, ds *datasources.DataSou
 		}
 	}
 
+	proxyOpts := &sdkproxy.Options{}
```

Comment:
Nit: It shouldn't be a problem but maybe it's safer to keep `nil` in case the proxy is not enabled and set the value only if it's enabled:

```go
// proxyOpts := &sdkproxy.Options{}
if ds.JsonData != nil && ds.JsonData.Get("enableSecureSocksProxy").MustBool(false) {
	opts.ProxyOptions = &sdkproxy.Options{
		Enabled:  true,
		Timeouts: &sdkproxy.DefaultTimeoutOptions,
		Auth: &sdkproxy.AuthOptions{
			Username: ds.JsonData.Get("secureSocksProxyUsername").MustString(ds.UID),
		},
	}
	// rest of the code...
}
// opts.ProxyOptions = proxyOpts
```

Reply:
Updated!